tests/lib/nested.sh

```diff
@@ -0,0 +1,210 @@
+        exit 1
+    fi
+
+    PARAM_DISPLAY="-nographic"
+    PARAM_NETWORK="-net nic,model=virtio -net user,hostfwd=tcp::${SSH_PORT}-:22"
+    # TODO: do we need monitor port still?
+    PARAM_MONITOR="-monitor tcp:127.0.0.1:${MON_PORT},server,nowait"
+    PARAM_RANDOM="-object rng-random,id=rng0,filename=/dev/urandom -device virtio-rng-pci,rng=rng0"
+    PARAM_CPU=""
+    PARAM_TRACE="-d cpu_reset"
+    PARAM_LOG="-D ${WORK_DIR}/qemu.log"
+    PARAM_SERIAL="-serial file:${WORK_DIR}/serial.log"
+    PARAM_TPM=""
+
+    ATTR_KVM=""
+    if [ "$ENABLE_KVM" = "true" ]; then
+        ATTR_KVM=",accel=kvm"
+        # CPU can be defined just when kvm is enabled
+        PARAM_CPU="-cpu host"
+        # Increase the number of cpus used once the issue related to kvm and ovmf is fixed
+        # https://bugs.launchpad.net/ubuntu/+source/kvm/+bug/1872803
+        PARAM_SMP="-smp 1"
+    fi
+
+    # TODO: enable ms key booting for i.e. nightly edge jobs ?
+    OVMF_CODE=""
+    OVMF_VARS=""
+    if [ "${ENABLE_SECURE_BOOT:-false}" = "true" ]; then
+        OVMF_CODE=".secboot"
+    fi
+    if [ "${ENABLE_OVMF_SNAKEOIL:-false}" = "true" ]; then
+        OVMF_VARS=".snakeoil"
+    fi
+
+    mkdir -p "${WORK_DIR}/image/"
+    cp -f "/usr/share/OVMF/OVMF_VARS${OVMF_VARS}.fd" "${WORK_DIR}/image/OVMF_VARS${OVMF_VARS}.fd"
+    PARAM_BIOS="-drive file=/usr/share/OVMF/OVMF_CODE${OVMF_CODE}.fd,if=pflash,format=raw,unit=0,readonly -drive file=${WORK_DIR}/image/OVMF_VARS${OVMF_VARS}.fd,if=pflash,format=raw"
+    PARAM_MACHINE="-machine q35${ATTR_KVM} -global ICH9-LPC.disable_s3=1"
+
+    if [ "${ENABLE_TPM:-false}" = "true" ]; then
+        TPMSOCK_PATH="/var/snap/swtpm-mvo/current/swtpm-sock"
+        if [ "${SPREAD_BACKEND}" = "lxd-nested" ]; then
+            mkdir -p /tmp/qtpm
+            swtpm socket --tpmstate dir=/tmp/qtpm --ctrl type=unixio,path=/tmp/qtpm/sock --tpm2 -d -t
+            TPMSOCK_PATH="/tmp/qtpm/sock"
+        elif ! snap list swtpm-mvo > /dev/null; then
+            snap install swtpm-mvo --beta
+            retry=60
+            while ! test -S /var/snap/swtpm-mvo/current/swtpm-sock; do
+                retry=$(( retry - 1 ))
+                if [ $retry -le 0 ]; then
+                    echo "Timed out waiting for the swtpm socket. Aborting!"
+                    return 1
+                fi
+                sleep 1
+            done
+        fi
+        PARAM_TPM="-chardev socket,id=chrtpm,path=${TPMSOCK_PATH} -tpmdev emulator,id=tpm0,chardev=chrtpm -device tpm-tis,tpmdev=tpm0"
+    fi
+
+    PARAM_IMAGE="-drive file=${IMAGE_FILE},cache=none,format=raw,id=disk1,if=none -device virtio-blk-pci,drive=disk1,bootindex=1"
+
+    # Create the systemd unit for running the VM, the qemu parameter order is 
+    # important. We run all of the qemu logic through a script that systemd runs
+    # for an easier time escaping everything
+    SVC_NAME="nested-vm-$(systemd-escape "${SPREAD_JOB:-unknown}")"
+    SVC_SCRIPT="/tmp/nested-vm-$RANDOM.sh"
```

Comment:
Why not using `mktemp`?

Reply:
It would be nice to still have it named something easy to recognize so if you have to debug the script it generates, or wanting to inspect how the inner VM is started. Does this make sense or would you still prefer it being mktemp?

Reply:
You can give a template. `mktemp --tmpdir nested-vm-XXXXXXXXXX`

Reply:
Cool, i'll test it out!